src/decode.rs

```diff
@@ -120,13 +132,26 @@ pub fn decode<R: Read>(schema: &Schema, reader: &mut R) -> Result<Value, Error>
                 _ => Err(DecodeError::new("union index out of bounds").into()),
             }
         },
-        &Schema::Record { ref fields, .. } => fields
-            .iter()
-            .map(|field| decode(&field.schema, reader).map(|value| (field.name.clone(), value)))
-            .collect::<Result<Vec<(String, Value)>, _>>()
-            .map(|items| Value::Record(items)),
+        &Schema::Record { ref fields, .. } => {
+            // Benchmarks indicate ~10% improvement using this method.
+            // Putting it in the `Value::Record` initially may be overkill, but
+            // not using the Iterator API appears to perform better.
+            let mut records = Value::Record(Vec::new());
```

Comment:
I think I was getting overzealous about reducing the moves. Just switched it back to something more like `let mut items = Vec::new(); { ... }; Ok(Value::Record(items)` and the benchmarks show no significant difference. I'll commit the change because it is more clear that way. Thanks for catching this